.evergreen/run-oidc-tests-k8s.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -o xtrace   # Write all commands first to stderr
+set -o errexit  # Exit the script with error if any of the commands fail
+
+export K8S_DRIVERS_TAR_FILE=/tmp/node-mongodb-native.tgz
+cd ..
+tar -czf $K8S_DRIVERS_TAR_FILE src drivers-tools
+cd -
+bash $DRIVERS_TOOLS/.evergreen/auth_oidc/k8s/setup-pod.sh
+bash $DRIVERS_TOOLS/.evergreen/auth_oidc/k8s/run-self-test.sh
+export K8S_TEST_CMD="cd src && DRIVERS_TOOLS=${DRIVERS_TOOLS} ENVIRONMENT=k8s ./.evergreen/${SCRIPT}"
+source $DRIVERS_TOOLS/.evergreen/auth_oidc/k8s/secrets-export.sh
+bash $DRIVERS_TOOLS/.evergreen/auth_oidc/k8s/run-driver-test.sh
+bash $DRIVERS_TOOLS/.evergreen/auth_oidc/k8s/teardown-pod.sh
```

Comment:
Do we need to run the teardown in a `post` block in the evergreen config, to ensure that we always clean up k8s resources?

Reply:
I had a chat with Steve about this and it's not important where the pod teardown runs as it's local and the important thing is the teardown of the Atlas cluster which happens in the task group teardown.